### PR TITLE
Added Button Layout

### PR DESCRIPTION
**Pull request by moshfeu**

### CODE REVIEW

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+/**
+ * ButtonLayout
+ */
+class ButtonLayout extends React.PureComponent<ButtonLayoutProps> {
+  static displayName = 'ButtonLayout';
+
+  static propTypes = {
+    /** Makes the button disabled */
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: which button? :)
To be accurate, it provides a disabled theme.

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    disabled: bool,
+    /** Standard button onClick callback */
+    onClick: func,
+    /** Standard button onMouseEnter callback */
+    onMouseEnter: func,
+    /** Standard button onMouseLeave callback */
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: no button

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+class ButtonLayout extends React.PureComponent<ButtonLayoutProps> {
+  static displayName = 'ButtonLayout';
+
+  static propTypes = {
+    /** Makes the button disabled */
+    disabled: bool,
+    /** Standard button onClick callback */
+    onClick: func,
+    /** Standard button onMouseEnter callback */
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: no button

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    onMouseEnter: func,
+    /** Standard button onMouseLeave callback */
+    onMouseLeave: func,
+    /** Classes object */
+    classes: object.isRequired,
+    /** Any node to be rendered (usually text node) */
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: This is not true, usually the children are elements (`<a>` or `<Link>`)

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  render() {
+    const {onClick, children, classes, onMouseEnter, onMouseLeave, disabled} = this.props;
+
+    return (
+      <div
+        onClick={() => this.internalOnClick(onClick)}
+        className={classes.buttonLayout}
+        onMouseEnter={onMouseEnter}
+        onMouseLeave={onMouseLeave}
+        data-disabled={disabled}
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: remove please, this should only be used for themeing. See how the disabled prop is handled in the ToggleSwitch

File: packages/wix-ui-core/src/components/ButtonLayout/styles.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import {core, ButtonLayoutTheme} from './theme';
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/styles.ts]: Don't duplicate this file, it should be identical to `Button/styles.ts` so use it

File: packages/wix-ui-core/src/components/ButtonLayout/theme.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import {palette} from '../../palette';
```
Comment (lbelinsk) [inline packages/wix-ui-core/src/components/ButtonLayout/theme.ts]: this is also a duplication of the Button theme, use the Button's one

File: packages/wix-ui-core/stories/index.tsx [hunk 8]
```diff
@@ -16,6 +17,9 @@ storiesOf('Components', module)
   .add('Button', () => (
     <Button dataHook="story-button">Hello</Button>
   ))
+  .add('ButtonLayout', () => (
+    <ButtonLayout dataHook="story-button-layout">Hello</ButtonLayout>
```
Comment (lbelinsk) [inline packages/wix-ui-core/stories/index.tsx]: Render an Anchor please as a children

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.driver.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+  return {
+    element: () => element,
+    exists: () => !!element,
+    click: () => eventTrigger.click(element),
+    mouseEnter: () => eventTrigger.mouseEnter(element),
+    mouseLeave: () => eventTrigger.mouseLeave(element),
+    getTextContent: () => element.textContent,
+    getHtmlContent: () => element.innerHTML,
+    isDisabled: () => element.getAttribute('data-disabled') === 'true',
```
[the remark this answers is not in the document]
Comment (moshfeu) [reply packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.driver.ts]: Due we can't set `disabled` attribute on NON `input` element (we are using `div`), we need to use `data-*` attribute.

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.driver.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    styles: {
+      /** returns elements height css property */
+      getHeight: () => getCssValue('height'),
```
[the remark this answers is not in the document]
Comment (moshfeu) [reply packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.driver.ts]: We created an internal method called `getCssValue` too.

File: packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx (hunk 5 above)
Comment (moshfeu) [reply packages/wix-ui-core/src/components/ButtonLayout/ButtonLayout.tsx]: Due we can't set disabled attribute on NON input element (we are using div), we need to use data-* attribute.

**Comment by shlomitc** [comment]
should be a stylable version
also, I have doubts again regarding the use of it as a separate component.

let's discuss it again when relevant

**Pull request closed without merging**